src/SHermitianCompact.jl

```diff
@@ -153,9 +153,19 @@ end
 
 @inline Base.:*(a::Number, b::SHermitianCompact) = SHermitianCompact(a * b.lowertriangle)
 @inline Base.:*(a::SHermitianCompact, b::Number) = SHermitianCompact(a.lowertriangle * b)
+@inline Base.:*(a::Complex, b::SHermitianCompact) = a * SMatrix(b)
+@inline Base.:*(a::SHermitianCompact, b::Complex) = SMatrix(a) * b
```

Comment:
Maybe it would be better to make your code the default for `Number` and return `SHermitianCompact` only when the number argument is `Real`.